src/Engine/Manager/RenderManager.cpp

```diff
@@ -203,82 +230,124 @@ void RenderManager::UpdateBufferSize() {
 void RenderManager::RenderWorldEntities(World& world, const glm::mat4& viewMatrix, const glm::mat4& projectionMatrix, Video::RenderSurface* renderSurface) {
     // Render from camera.
     renderer->StartRendering(renderSurface);
+    glm::mat4 lightViewMatrix;
+    glm::mat4 lightProjection;
+    float aspectRatio = static_cast<float>(shadowPass->GetShadowWidth()) / shadowPass->GetShadowHeight();
+
+    for (Component::SpotLight* spotLight : spotLights.GetAll()) {
+        if (spotLight->IsKilled() || !spotLight->entity->enabled)
+            continue;
+
+        if (spotLight->shadow) {
+            Entity* lightEntity = spotLight->entity;
+            lightViewMatrix = glm::inverse(lightEntity->GetModelMatrix());
+            lightProjection = glm::perspective(glm::radians(2.f * spotLight->coneAngle), aspectRatio, 0.5f, 50.0f);
+        }
+    }
 
     // Camera matrices.
     const glm::mat4 viewProjectionMatrix = projectionMatrix * viewMatrix;
-
     const std::vector<Mesh*>& meshComponents = meshes.GetAll();
-
+    //Render shadows.
```

Comment:
The actual shadows aren't rendered. We're rendering the shadow map.

Reply:
Fixed